tests/ImageSharp.Tests/TestUtilities/ImageProviders/TestPatternProvider.cs

```diff
@@ -0,0 +1,184 @@
+﻿// <copyright file="BlankProvider.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Numerics;
+
+    public abstract partial class TestImageProvider<TColor>
```

Comment:
Can I get some descriptions on the private methods here. Just to ensure everyone can easily see exactly what we are testing.